### PR TITLE
feat: [ion_identity_client] added password register flow

### PR DESCRIPTION
**Pull request by ice-hades**
## Description
Implemented register a new user with password if passkey is not available flow

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Refactoring
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

File: packages/ion_identity_client/lib/src/auth/services/register/register_service.dart [hunk 1]
```diff
@@ -24,28 +28,40 @@ class RegisterService {
...
+      credentialData: credentialData,
       temporaryAuthenticationToken: userRegistrationChallenge.temporaryAuthenticationToken,
     );
     await tokenStorage.setTokens(
       username: username,
       newTokens: registrationCompleteResponse.authentication,
     );
   }
+
+  Future<bool> isPasskeyAuthAvailable() async {
```
Comment (ice-tychon) [inline packages/ion_identity_client/lib/src/auth/services/register/register_service.dart:64]: It shouldn't belong here.

File: packages/ion_identity_client/lib/src/auth/dtos/credential_request_data.dart [hunk 2]
```diff
@@ -6,12 +6,15 @@ import 'package:json_annotation/json_annotation.dart';
 
 part 'credential_request_data.g.dart';
 
+// ignore: constant_identifier_names
+enum CredentialKind { Fido2, Key, PasswordProtectedKey, RecoveryKey }
+
 @JsonSerializable()
 class CredentialRequestData {
```
Comment (ice-tychon) [inline packages/ion_identity_client/lib/src/auth/dtos/credential_request_data.dart:13]: We use this class for a recovery creds request. It shouldn't be used for register.

File: packages/ion_identity_client/lib/src/signer/password_signer.dart [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+import 'dart:convert';
+
+import 'package:convert/convert.dart';
+import 'package:crypto/crypto.dart';
+import 'package:cryptography/cryptography.dart' as crypto;
+import 'package:ion_identity_client/ion_identity.dart';
+import 'package:ion_identity_client/src/auth/dtos/dtos.dart';
+import 'package:ion_identity_client/src/auth/services/key_service.dart';
+
+class PasswordSigner {
```
Comment (ice-tychon) [inline packages/ion_identity_client/lib/src/signer/password_signer.dart:12]: Let's create a Facade for signers. Injecting multiple signers into a service feels messy. A Facade will simplify access and improve maintainability.

File: packages/ion_identity_client/lib/src/signer/password_signer.dart [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  PasswordSigner({
+    required this.config,
+    required this.keyService,
+  });
+
+  final KeyService keyService;
+  final IONIdentityConfig config;
+
+  Future<CredentialRequestData> getCredentialInfo({
```
Comment (ice-tychon) [inline packages/ion_identity_client/lib/src/signer/password_signer.dart:21]: The class name is also too specific. From what I see, it signs more than just the `PasswordProtectedKey`.

**Review by ice-alcides — APPROVED** [review]

**Pull request merged**